app/controllers/alarms_controller.rb

```diff
@@ -1,71 +1,51 @@
 class AlarmsController < ApplicationController
   def index
-    @alarms = current_user.alarms
-    @alarms = @alarms.map do |alarm|
-      [alarm, alarm.alarmDays]
-    end
-    @alarms
+    @alarms = current_user.alarms.map {|alarm| helpers.decorate(alarm)}
   end
 
   def new
     @alarm = Alarm.new
-    @user = current_user
-    @days = create_days
   end
 
   def create
-    if params[:days].empty?
-      # message of invalid
-    else 
-      @alarm = current_user.alarms.create(alarm_params)
-      if @alarm
-        hours, minutes = params[:alarm][:time].split(':')
-        params[:days].each do |day|
-          new_time = DateTime.parse(day).change(hour: hours.to_i, min: minutes.to_i)
-          @alarm.alarmDays.create({daytime: new_time})
-          AlarmNotificationJob.set(wait_until: new_time).perform_later(current_user, @alarm)
-        end
-        redirect_to user_alarms_path(current_user)
-      else
-        render :new
+    @alarm = current_user.alarms.create(alarm_params)
+    if @alarm
+      hours, minutes = params[:alarm][:time].split(':')
+      alarm_params[:days].each do |day|
+        new_time = DateTime.parse(day).change(hour: hours.to_i, min: minutes.to_i)
+        AlarmNotificationJob.set(wait_until: new_time).perform_later(current_user, @alarm)
```

Comment:
Maybe you could implement a service to handle this logic, so your `create` action would be cleaner